Find order

adityasharma7 opened the pull request.

ymaheshwari1 committed:
Implemented an ionic modal component for displaying order filter on mobile(#22qhq27)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into #22qhq27

ymaheshwari1 committed:
Updated: the static UI as per the figma and added the static text in the locale(#22qhq27)

ymaheshwari1 committed:
Merge branch '#1x634dq' into #22qhq27

ymaheshwari1 committed:
Added: appliedFilter object to store the currently selected filters(#22qhq27)

ymaheshwari1 committed:
Added: condition to update the solr-query filters when any selected filter is changed(#22qhq27)

ymaheshwari1 committed:
Added: faceting logic and made the fulfillment filters functional(#22qhq27)

ymaheshwari1 committed:
Updated: solr-query to display the latest orders first(#22qhq27)

ymaheshwari1 committed:
Added: check for display productName, color, size and brandName(#22qhq27)

ymaheshwari1 committed:
Implemented: action to fetch and display the PO ids in the filter(#22qhq27)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into #22qhq27

ymaheshwari1 committed:
Added: a check for the filter to not return null or undefined(#22qhq27)

ymaheshwari1 committed:
Added: interface popover option on ion-select component(#22qhq27)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into #22qhq27

ymaheshwari1 committed:
Updated: code to correctly prepare the solr-query when applying multiple filters(#22qhq27)

ymaheshwari1 committed:
Added: any as default value for the fulfillment filters(#22qhq27)

ymaheshwari1 committed:
Added: a condition to check for item status or order status in env file(#22qhq27)

ymaheshwari1 committed:
Added: datetime component to select date and filter the result based on the date(#22qhq27)

ymaheshwari1 committed:
Changed: the click event from ion-list to individual ion-checkbox(#22qhq27)

ymaheshwari1 committed:
Updated: the date logic in solr-query to get the orders(#22qhq27)

ymaheshwari1 committed:
Added: changelog entry file(#22qhq27)

ymaheshwari1 committed:
Changed: click event to ionChange event for ionCheckbox(#22qhq27)

ymaheshwari1 committed:
Implemented: state for storing and displaying the filter options(#22xecka)

ymaheshwari1 committed:
Implemented: state for storing applied filters and defined an action to prepare the query(#22xecka)

ymaheshwari1 committed:
Updated: the parameter passing logic(#22xecka)

ymaheshwari1 committed:
Updated: the filter modal for mobile device to display the datetime component in a modal(#22xecka)

ymaheshwari1 committed:
Added: ion-icon on the date chip to make it to default value(#22xecka)

ymaheshwari1 committed:
Updated: the en.json file(#22xecka)

ymaheshwari1 committed:
Removed: the sync and download csv button from UI(#22xecka)

ymaheshwari1 committed:
Improved: name for ion-modal id and used service instead of using an action for getting po ids(#22qhq27)

ymaheshwari1 committed:
Improved: code to use ion-menu for display filters on mobile device and removed the usage of ion-datetime and used ion-input(#22xecka)

ymaheshwari1 committed:
Added: changelog entry file(#22xecka)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into #22qhq27

ymaheshwari1 committed:
Improved: order object as per the order type defined(#22qhq27)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into #22qhq27

dt2patel committed:
Merge pull request #66 from ymaheshwari1/#22qhq27

Implemented functionality to filter the orders on the basis of filter selected(#22qhq27)

ymaheshwari1 committed:
Merge branch 'FindOrder' into #22xecka

dt2patel committed:
Merge pull request #71 from ymaheshwari1/#22xecka

Implemented: order filters for mobile device and updated the logic to store the filters in the state(#22xecka)

ymaheshwari1 committed:
Implemented functionality to display PO ids and filter order on the basis of PO ids(#2335pf5)

ymaheshwari1 committed:
Updated: release changelog file(#2335pf5)

ymaheshwari1 committed:
Fixed: issue of filters always passed as empty value in api call(#2335pf5)

ymaheshwari1 committed:
Implemented: condition to display selected po ids(#2335pf5)

ymaheshwari1 committed:
Improved: code to use join method instead of using replaceAll(#2335pf5)

ymaheshwari1 committed:
Improved: code to move the logic of po filter update to actions(#2335pf5)

ymaheshwari1 committed:
Updated: method name from availableFilterOptionsUpdated to updateAvailableFilterOptions(#2335pf5)

ymaheshwari1 committed:
Implemented: functionality to sort orders on the basis of date and added option to show or hide the order items(#270yrt1)

ymaheshwari1 committed:
Updated: the relase changelog file(#270yrt1)

ymaheshwari1 committed:
Improved: method name from appliedFiltersUpdated to updateAppliedFilters(#2335pf5)

ymaheshwari1 committed:
Improved: code to filter empty result using solr query(#2335pf5)

adityasharma7 committed:
Merge pull request #94 from ymaheshwari1/#2335pf5-update

Implemented functionality to display PO ids and filter order on the basis of PO ids(#2335pf5)

ymaheshwari1 committed:
Merge branch 'FindOrder' of https://github.com/hotwax/commerce-hub into #270yrt1

ymaheshwari1 committed:
Updated: code to import the used icons(#270yrt1)

ymaheshwari1 committed:
Improved: code to store the sort option with sort direction(#270yrt1)

ymaheshwari1 committed:
Changed: variable name from promiseDate to promisedDate(#270yrt1)

adityasharma7 committed:
Merge pull request #97 from ymaheshwari1/#270yrt1

Implemented: functionality to sort orders on the basis of date and added option to show or hide the order items(#270yrt1)

ymaheshwari1 committed:
Implemented: solr helper to prepare order query(#2712605)

ymaheshwari1 committed:
Updated: order module for handling action call and preparing solr logic(#2712605)

ymaheshwari1 committed:
Improved: the code for the UI components as per the updated state(#2712605)

ymaheshwari1 committed:
Updated: the order state to not store the filter options and store the poIds, also update the variable name and improved the logic as per the updated state(#2712605)

ymaheshwari1 committed:
Removed: unwanted console statement(#2712605)

ymaheshwari1 committed:
Removed: unwanted state from order module(#2712605)

ymaheshwari1 committed:
Updated: release changelog file(#2712605)

adityasharma7 committed:
Merge pull request #98 from ymaheshwari1/#2712605

Implemented: solr helper to prepare order query and updated the state for storing filters(#2712605)

ymaheshwari1 committed:
Merge branch 'main' of https://github.com/hotwax/commerce-hub into FindOrder

ymaheshwari1 committed:
Implemented: state in order module to store possible valid status change option and to change the order status(#2335qmv)

ymaheshwari1 committed:
Updated: the persist state paths to include the valid status change options(#2335qmv)

ymaheshwari1 committed:
Implemented: service to fetch and update the order status and called the action to fetch valid status after login(#2335qmv)

ymaheshwari1 committed:
Updated: the release changelog file(#2335qmv)

ymaheshwari1 committed:
Fixed: the issue of searchbar text is not passed correctly in the api when searching too fast(#2335qmv)

ymaheshwari1 committed:
Updated: the entity name to change order status and passed setItemStatus(#2335qmv)

ymaheshwari1 committed:
Improved: condition to not set the value for ion-select and added condition to not show the order status change options when we are having an empty list(#2335qmv)

ymaheshwari1 committed:
Removed: unwanted comment(#2335qmv)

adityasharma7 committed:
Merge pull request #99 from ymaheshwari1/#2335qmv

Implemented functionality to change the order status(#2335qmv)

ymaheshwari1 committed:
Updated: code to pass the query string when applying a filter(#2335qmv)

ymaheshwari1 committed:
Merge branch 'FindOrder' of https://github.com/hotwax/commerce-hub into FindOrder

adityasharma7 commented on src/components/OrderFilters.vue:
```diff
@@ -0,0 +1,158 @@
+    <ion-item>
+      <ion-label>{{ $t("Status") }}</ion-label>
+      <ion-select :value="query.status" @ionChange="updateAppliedFilters($event['detail'].value, 'status')" interface="popover">
+        <ion-select-option v-for="status in orderStatusOptions" :key="status" :value="status">{{ status }}</ion-select-option>
```
I think value should be id and label should be the description

ymaheshwari1 replied:
Sir, in orderStatusOptions we are only getting the id as we are fetching the info from solr facets. So for displaying the label used the env file entry.

adityasharma7 commented on src/components/OrderFilters.vue:
```diff
@@ -0,0 +1,158 @@
+    <ion-item>
+      <ion-label>{{ $t("Shipping method") }}</ion-label>
+      <ion-select :value="query.shippingMethod" @ionChange="updateAppliedFilters($event['detail'].value, 'shippingMethod')" interface="popover">
+        <ion-select-option v-for="method in shippingMethodOptions" :key="method" :value="method">{{ method }}</ion-select-option>
```
I think value should be id and label should be the description

ymaheshwari1 replied:
Sir, we are fetching the shipping method options from solr facets so we are only getting the id, so used the shipping method getter to display the desc for the shipping method.

adityasharma7 commented on src/components/OrderFilters.vue:
```diff
@@ -0,0 +1,158 @@
+    </ion-item>
+    <ion-item>
+      <ion-label>{{ $t("Ship from location") }}</ion-label>
+      <ion-select :value="query.shipFromLocation" @ionChange="updateAppliedFilters($event['detail'].value, 'shipFromLocation')" interface="popover">
```
I think here we should use facilityTypeId values

ymaheshwari1 replied:
Sir, used facilityTypeId as a hardcoded value.

adityasharma7 commented on src/components/OrderFilters.vue:
```diff
@@ -0,0 +1,158 @@
+  },
+  methods: {
+    async updateAppliedFilters(value: string, filterName: string) {
+      if (value === this.query[filterName] && !(filterName === 'orderCreated' || filterName === 'promisedDate' || filterName === 'autoCancelDate')) {
```
I think this should not be needed. Please verify

adityasharma7 commented on src/store/modules/order/actions.ts:
```diff
@@ -30,6 +41,7 @@ const actions: ActionTree<OrderState, RootState> = {
         const total = resp.data.grouped.orderId.ngroups;
         if (payload.json.params.start && payload.json.params.start > 0) orders = state.list.orders.concat(orders)
         this.dispatch('product/getProductInformation', { orders });
+        this.dispatch('order/fetchStatusChange')
```
Why this is needed on every time order is fetch?

ymaheshwari1 replied:
Sir, it's not needed on orders fetch, I just added it for testing, will remove it.

adityasharma7 commented on src/store/modules/order/actions.ts:
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
+    return resp;
+  },
+
+  async updateQueryString({ commit }, queryString) {
```
I think this is not needed, We could use `updateAppliedFilters`

ymaheshwari1 replied:
Done Sir

adityasharma7 commented on src/store/modules/order/actions.ts:
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
+      payload.value = poIds
+    }
+    commit(types.ORDER_FILTERS_UPDATED, payload)
+    const resp = await dispatch('updateQuery', { viewSize: process.env.VUE_APP_VIEW_SIZE, viewIndex: 0, queryString: state.query.queryString })
```
If queryString is already in state no need to pass it

ymaheshwari1 replied:
Done Sir

adityasharma7 commented on src/store/modules/order/actions.ts:
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
+  },
+
+  async updateQuery({ state, dispatch }, params) {
+    await dispatch('updateQueryString', params.queryString)
```
I think we should update query in applied filters and dispatch updateQuery instead

ymaheshwari1 replied:
Done Sir

adityasharma7 commented on src/views/FindOrder.vue:
```diff
@@ -332,11 +278,37 @@ export default defineComponent ({
         event.target.complete();
       })
+    },
+    async openOrderFilterMenu() {
```
```suggestion
    async openOrderFilter() {
```

ymaheshwari1 replied:
Improved the method name

adityasharma7 commented on src/store/modules/order/actions.ts:
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
+    return resp;
+  },
+
+  async updateSortOption({ commit }, payload) {
```
```suggestion
  async updateSort({ commit }, payload) {
```

ymaheshwari1 replied:
Updated the method name

adityasharma7 commented on src/views/FindOrder.vue:
```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
+    async sortOrders(value: string) {
+      this.sort = value
+      await this.store.dispatch('order/updateSortOption', this.sort)
+      this.getOrders();
```
I think we could this in updateSort action

ymaheshwari1 replied:
Moved the getOrders call to actions.

adityasharma7 commented on src/views/FindOrder.vue:
```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
-        payload.json.params['q.op'] = 'AND'
-        payload.json.query = `*${this.queryString}*`
+
+      const resp = await this.store.dispatch('order/updateQuery', { viewSize, viewIndex, queryString: this.queryString })
```
If we are already setting the queryString no need to pass it again

ymaheshwari1 replied:
Done Sir

adityasharma7 commented on src/views/FindOrder.vue:
```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
-        payload.json.params['q.op'] = 'AND'
-        payload.json.query = `*${this.queryString}*`
+
+      const resp = await this.store.dispatch('order/updateQuery', { viewSize, viewIndex, queryString: this.queryString })
```
Have separate method for infinite scroll, pass only in that case and skip passing for other cases. updateQuery action should only handle if viewSize is passed with default values

ymaheshwari1 replied:
Improved the code on infinite scroll.

ymaheshwari1 committed:
Improved: code to display the label for status and shipping method options(FindOrder)

ymaheshwari1 committed:
Removed: the order valid status change action call on order fetch action(FindOrder)

ymaheshwari1 committed:
Updated: the methods names as per the suggestion(FindOrder)

ymaheshwari1 committed:
Improved: code to simplify the order query logic(FindOrder)

ymaheshwari1 committed:
Improved: code to save the id for ship locations as select values(FindOrder)

ymaheshwari1 committed:
Improved: code to display the shipment method description(FindOrder)

ymaheshwari1 committed:
Removed: unwanted action defination(FindOrder)

adityasharma7 commented on src/store/modules/order/index.ts:
```diff
@@ -12,7 +12,25 @@ const orderModule: Module<OrderState, RootState> ={
+    query: {
+      'status': 'any',
+      'shippingMethod': 'any',
+      'shipFromLocation': 'any',
```
```suggestion
      'shipFromLocation': '',
```

adityasharma7 committed:
Improved: used empty initial value instead of any (#22qhq27)

Any is not any speicific location, we could handle it with empty string.

adityasharma7 commented on src/components/OrderFilters.vue:
```diff
@@ -0,0 +1,161 @@
+    <ion-item>
+      <ion-label>{{ $t("Ship from location") }}</ion-label>
+      <ion-select :value="query.shipFromLocation" @ionChange="updateAppliedFilters($event['detail'].value, 'shipFromLocation')" interface="popover">
+        <ion-select-option value="any" >{{ $t('any') }}</ion-select-option>
```
```suggestion
        <ion-select-option value="" >{{ $t('any') }}</ion-select-option>
```

adityasharma7 committed:
Improved: used empty initial value instead of any (#22qhq27)

Any is not any speicific location, we could handle it with empty string.

adityasharma7 commented on src/utils/solrHelper.ts:
```diff
@@ -0,0 +1,107 @@
+
+  payload.json.filter = payload.json.filter.concat(` AND facilityId: (${typeFilterValues ? typeFilterValues : '*'})`)
+
+  if (query.shipFromLocation !== 'any') {
```
```suggestion
  if (query.shipFromLocation) {
```

adityasharma7 committed:
Improved: used empty initial value instead of any (#22qhq27)

Any is not any specific location, we could handle it initialising with empty string.

ymaheshwari1 committed:
Removed: usage of state directly in solrHelper()

adityasharma7 committed:
Fixed: log warnings (#22qhq27)

adityasharma7 committed:
Merge branch 'FindOrder' of https://github.com/hotwax/commerce-hub into FindOrder

adityasharma7 committed:
Improved: code to initially set empty string instead of any (#22qhq27)
Handling empty string would be much easier than add extra checks for any
